build_runner_core/lib/src/changes/build_script_updates.dart

```diff
@@ -13,14 +13,26 @@ import '../asset/reader.dart';
 import '../asset_graph/graph.dart';
 import '../package_graph/package_graph.dart';
 
+/// Functionality for detecting if the build script itself or any of its
+/// transitive imports have changed.
 class BuildScriptUpdates {
   final Set<AssetId> _allSources;
   final bool _supportsIncrementalRebuilds;
 
   BuildScriptUpdates._(this._supportsIncrementalRebuilds, this._allSources);
 
-  static Future<BuildScriptUpdates> create(RunnerAssetReader reader,
-      PackageGraph packageGraph, AssetGraph graph) async {
+  /// Creates a [BuildScriptUpdates] object, using [reader] to ensure that
+  /// the [graph] is tracking digests for all transitive sources.
+  ///
+  /// If [skipUpdatesCheck] is `true` then all checks are skipped and
+  /// [hasBeenUpdated] will always return `false`.
+  static Future<BuildScriptUpdates> create(
+      RunnerAssetReader reader,
+      PackageGraph packageGraph,
+      AssetGraph graph,
+      bool skipUpdatesCheck) async {
```

Comment:
how about a named argument?

Also, maybe call it `{bool disabled = false}`?

Reply:
done